internal/network/command_create.go

```diff
@@ -0,0 +1,189 @@
+		return err
+	}
+
+	connectionTypes, err := cmd.Flags().GetStringSlice("connection-types")
+	if err != nil {
+		return err
+	}
+
+	for i, connectionType := range connectionTypes {
+		connectionTypes[i] = strings.ToUpper(connectionType)
+	}
+
+	cidr, err := cmd.Flags().GetString("cidr")
+	if err != nil {
+		return err
+	}
+
+	zones, err := cmd.Flags().GetStringSlice("zones")
+	if err != nil {
+		return err
+	}
+
+	zoneInfo, err := cmd.Flags().GetStringSlice("zone-info")
+	if err != nil {
+		return err
+	}
+	zoneInfoItems, err := getZoneInfoItems(zoneInfo)
+	if err != nil {
+		return err
+	}
+
+	dnsResolution, err := cmd.Flags().GetString("dns-resolution")
+	if err != nil {
+		return err
+	}
+	dnsResolution, err = formatDnsResolution(dnsResolution)
+	if err != nil {
+		return err
+	}
+
+	reservedCidr, err := cmd.Flags().GetString("reserved-cidr")
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	createNetwork := networkingv1.NetworkingV1Network{
+		Spec: &networkingv1.NetworkingV1NetworkSpec{
+			DisplayName:     networkingv1.PtrString(name),
+			Cloud:           networkingv1.PtrString(cloud),
+			Region:          networkingv1.PtrString(region),
+			ConnectionTypes: &networkingv1.NetworkingV1ConnectionTypes{Items: connectionTypes},
+			Environment:     &networkingv1.ObjectReference{Id: environmentId},
+		},
+	}
+
+	if cidr != "" {
+		createNetwork.Spec.SetCidr(cidr)
+	}
+
+	if len(zones) != 0 {
+		createNetwork.Spec.SetZones(zones)
+	}
+
+	if len(zoneInfoItems) != 0 {
+		createNetwork.Spec.SetZonesInfo(networkingv1.NetworkingV1ZonesInfo{Items: zoneInfoItems})
+	}
+
+	if dnsResolution != "" {
+		createNetwork.Spec.SetDnsConfig(networkingv1.NetworkingV1DnsConfig{Resolution: dnsResolution})
+	}
+
+	if reservedCidr != "" {
+		createNetwork.Spec.SetReservedCidr(reservedCidr)
+	}
+
+	network, err := c.V2Client.CreateNetwork(createNetwork)
+	if err != nil {
+		return err
+	}
+
+	return printTable(cmd, network)
+}
+
+func getZoneInfoItems(zoneInfo []string) ([]networkingv1.NetworkingV1ZoneInfo, error) {
+	zoneInfoItems := make([]networkingv1.NetworkingV1ZoneInfo, len(zoneInfo))
+	for i, info := range zoneInfo {
+		zoneInfoSplit := strings.Split(info, "=")
+		if len(zoneInfoSplit) == 1 {
+			zoneInfoItems[i] = networkingv1.NetworkingV1ZoneInfo{
+				Cidr: networkingv1.PtrString(zoneInfoSplit[0]),
+			}
+		} else if len(zoneInfoSplit) == 2 {
+			zoneInfoItems[i] = networkingv1.NetworkingV1ZoneInfo{
+				ZoneId: networkingv1.PtrString(zoneInfoSplit[0]), Cidr: networkingv1.PtrString(zoneInfoSplit[1]),
+			}
+		} else {
+			return nil, fmt.Errorf(`zone info "%s" is not correctly formatted as <zone-ID>=<CIDR> or <CIDR>`, info)
+		}
+	}
+	return zoneInfoItems, nil
+}
+
+func formatDnsResolution(dnsResolution string) (string, error) {
+	switch dnsResolution {
+	case "":
+		return "", nil
+	case "private":
+		return "PRIVATE", nil
+	case "chased-private":
+		return "CHASED_PRIVATE", nil
```

Comment:
Typically we upper-case and replace "-" with "_" in case we want to add a new type in the backend without needing to update the CLI